File: WinGo/src/main/java/Entites/Tag.java
package Entites;

import java.util.Objects;

public class Tag {
    private int id;
    private String nom;

    public Tag() {}

    public Tag(String nom) {
        this.nom = nom;
    }

    public Tag(int id, String nom) {
        this.id = id;
        this.nom = nom;
    }

    public int getId() { return id; }
    public void setId(int id) { this.id = id; }

    public String getNom() { return nom; }
    public void setNom(String nom) { this.nom = nom; }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        Tag tag = (Tag) o;
        return id == tag.id || Objects.equals(nom, tag.nom);
    }

    @Override
    public int hashCode() {
        return Objects.hash(id, nom);
    }

    @Override
    public String toString() {
        return "#" + nom;
    }
}

File: src/Entity/Tag.php
<?php

namespace App\Entity;

use Doctrine\ORM\Mapping as ORM;

use Doctrine\Common\Collections\Collection;
use App\Entity\Article_tag;

#[ORM\Entity]
class Tag
{

    #[ORM\Id]
    #[ORM\Column(type: "integer")]
    private int $id;

    #[ORM\Column(type: "string", length: 50)]
    private string $nom;

    public function getId()
    {
        return $this->id;
    }

    public function setId($value)
    {
        $this->id = $value;
    }

    public function getNom()
    {
        return $this->nom;
    }

    public function setNom($value)
    {
        $this->nom = $value;
    }

    #[ORM\OneToMany(mappedBy: "tag_id", targetEntity: Article_tag::class)]
    private Collection $article_tags;

        public function getArticle_tags(): Collection
        {
            return $this->article_tags;
        }
    
        public function addArticle_tag(Article_tag $article_tag): self
        {
            if (!$this->article_tags->contains($article_tag)) {
                $this->article_tags[] = $article_tag;
                $article_tag->setTag_id($this);
            }
    
            return $this;
        }
    
        public function removeArticle_tag(Article_tag $article_tag): self
        {
            if ($this->article_tags->removeElement($article_tag)) {
                // set the owning side to null (unless already changed)
                if ($article_tag->getTag_id() === $this) {
                    $article_tag->setTag_id(null);
                }
            }
    
            return $this;
        }
}
